sources/packages/api/src/services/atbc/atbc.service.ts

```diff
@@ -0,0 +1,180 @@
+   * @returns the result of a success full authentication or throws an exception
+   * in case the result is anything different from HTTP 200 code.
+   */
+  public async ATBCLogin(): Promise<ATBCAuthTokenResponse> {
+    try {
+      const agent = new (require("https").Agent)({
+        rejectUnauthorized: false,
+      });
+
+      const authRequest = await axios.post(
+        this.config.ATBCloginEndpoint,
+        {
+          usr: this.config.ATBCUserName,
+          pwd: this.config.ATBCPassword,
+          app: this.config.ATBCApp,
+        },
+        { httpsAgent: agent },
+      );
+      return authRequest?.data as ATBCAuthTokenResponse;
+    } catch (excp) {
+      this.logger.error(`Received exception while ATBC Authentication`);
+      this.logger.error(`${JSON.stringify(excp.response)} `);
+      throw excp;
+    }
+  }
+
+  /**
+   * Create client on ATBC.
+   * @returns the result of a success full authentication or throws an exception
+   * in case the result is anything different from HTTP 200 code.
+   */
+  public async ATBCCreateClient(
+    payload: ATBCCreateClientPayload,
+  ): Promise<ATBCCreateClientResponse> {
+    try {
+      const config = await this.getConfig();
+      const apiEndpoint =
+        this.config.ATBCEndpoint.slice(-1) === "/"
+          ? `${this.config.ATBCEndpoint}pd-clients`
+          : `${this.config.ATBCEndpoint}/pd-clients`;
+      const res = await axios.post(`${apiEndpoint}`, payload, config);
+      return res?.data as ATBCCreateClientResponse;
+    } catch (excp) {
+      this.logger.error(`Received exception while creating client at ATBC`);
+      this.logger.error(excp);
+      throw excp;
+    }
+  }
+  /**
+   * Check PD status of the student with SIN number on ATBC.
+   * @returns the result of a success full authentication or throws an exception
+   * in case the result is anything different from HTTP 200 code.
+   */
+  private async ATBCPDChecker(
+    payload: ATBCPDCheckerPayload,
+  ): Promise<ATBCPDCheckerResponse> {
+    try {
+      const config = await this.getConfig();
+      const apiEndpoint =
+        this.config.ATBCEndpoint.slice(-1) === "/"
+          ? `${this.config.ATBCEndpoint}pd`
+          : `${this.config.ATBCEndpoint}/pd`;
+      const res = await axios.post(`${apiEndpoint}`, payload, config);
+      return res?.data as ATBCPDCheckerResponse;
+    } catch (excp) {
+      this.logger.error(
+        `Received exception while checking the PD status at ATBC`,
+      );
+      this.logger.error(excp);
+      throw excp;
+    }
+  }
+
+  public async PDCheckerProcess(eachStudent: Student) {
+    try {
+      this.logger.log(`Creating Payload for student id ${eachStudent.id}`);
+      // create PD checker payload
+      const payload: ATBCPDCheckerPayload = {
+        id: eachStudent.sin,
+      };
+      // api to check the student PD status in ATBC
+      this.logger.log(`Check for PD status of student ${eachStudent.id}`);
+      // try {
```

Comment:
Is this some leftover comment?

Reply:
no, maybe i can edit the comment like, ``Check starting for PD status of student ${eachStudent.id}``

Reply:
Yes, make sense.